File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string log) {
        int message_pos = log.find(": ");
        return log.substr(message_pos + 2);
    }

    std::string log_level(std::string log) {
        int level_end_pos = log.find("]");
        return log.substr(1, level_end_pos - 1);
    }

    std::string reformat(std::string log) {
        return message(log) + " (" + log_level(log) +")";
    }
} // namespace log_line

File: solutions/csharp/log-levels/2/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        return logLine.Split(": ")[1].Trim();
    }

    public static string LogLevel(string logLine)
    {
        return logLine.Split("]")[0].Substring(1).ToLower();
    }

    public static string Reformat(string logLine)
    {
        string message = Message(logLine);
        string log = LogLevel(logLine);                                
        return $"{message} ({log})";
    }
}
